docs/OnTrack/JPlag/Jplag-Feasibility-Document.md

```diff
@@ -0,0 +1,118 @@
+
+Adding another container to the docker_compose file in .dev_container will require an image that
+supports Java SE 21, such as eclipse-temurin:21-alpine. It will also reference the .jar file to
+create a volume in the container.
+
+The current **unit_similarity_module.rb** file already defines how the batch process is managed. I
+believe using this method will also be suitable for JPLAG. This file can be altered to use ruby shell commands to: start jplag container > run jplag jar file > remove jplag container.
+
+The current MOSS batch process **unit_similarity_module.rb** is called from a rake file task.
+
+## Backend
+
+---
+
+JPLAG lets you define a source code file as a template, so it won't report any plagiarised lines if
+they already exist in that file. There will need to be a way for the task source code file (if
+exists) to be sent to JPLAG when it's running on that task's submissions.
+
+When JPLAG generates a report, it comes in a zip file with the following structure:
+
+```
+result.zip
+│   overview.json
+|   submissionFileIndex.json
+|   options.json
+│
+└───submissions
+│   └───submissionId1
+│       │   submission1File1
+│       │   submission1File2
+│       │   ...
+│   └───submissionId...
+│       │   ...
+│   └───submissionIdN
+│       │   submissionNFile1
+│       │   submissionNFile2
+│       │   ...
+│   submission1-submission2.json
+│   submission1-submission3.json
+│   submission1-submission....json
+│   submission1-submissionN.json
+│   ...
+```
+
+We still need to decide how we will manage the storage of these reports, because obviously we don't
+want to take up space storing the code files twice.
+
+We should be able to use the information defined in the comparison json files to populate the
+existing **task similarities table**. The comparison jsons give an average and max similarity
+percentage, which we will use the **max** to populate the **pct** field in the table.
+
+## Frontend
+
+---
+
+If you login as admin or convenor, navigate to any unit > Inbox > Explorer > goto any of student submission, currently you will find a "View Similarities" button for the MOSS plagiarism detection report. We aim to extend this functionality to include JPlag. To achieve this, we plan to embed the JPlag report viewer into this section for seamless integration. The report generated by JPlag, available in a zip file format, will be processed and displayed in a readable format similar to the existing MOSS reports.
+
+We have created a button labeled "Download JPLAG Report," which can be seen in the below UI; positioned next to the "View Similarities" button, allowing the teaching staff to download the detailed JPLAG analysis. This button will be accessible in the task explorer section (navigate to any unit > Inbox > Explorer) and will offer batch operations such as bulk export of submission files and PDFs. Once clicked, it will trigger the download of the generated JPLAG report zip file.
```

Comment:
`We have created a button labeled "Download JPLAG Report," which can be seen in the below UI; positioned next to the "View Similarities" button, allowing the teaching staff to download the detailed JPLAG analysis. `
This is also outdated